sql/planbuilder/ddl.go

```diff
@@ -0,0 +1,747 @@
+	return
+}
+
+func (b *PlanBuilder) buildConstraintsDefs(inScope *scope, tname ast.TableName, spec *ast.TableSpec) (fks []*sql.ForeignKeyConstraint, checks []*sql.CheckConstraint) {
+	for _, unknownConstraint := range spec.Constraints {
+		parsedConstraint := b.buildConstraintDefinition(inScope, unknownConstraint)
+		switch constraint := parsedConstraint.(type) {
+		case *sql.ForeignKeyConstraint:
+			constraint.Database = tname.Qualifier.String()
+			constraint.Table = tname.Name.String()
+			if constraint.Database == "" {
+				constraint.Database = b.ctx.GetCurrentDatabase()
+			}
+			fks = append(fks, constraint)
+		case *sql.CheckConstraint:
+			checks = append(checks, constraint)
+		default:
+			err := sql.ErrUnknownConstraintDefinition.New(unknownConstraint.Name, unknownConstraint)
+			b.handleErr(err)
+		}
+	}
+	return
+}
+
+func columnOrderToColumnOrder(order *ast.ColumnOrder) *sql.ColumnOrder {
+	if order == nil {
+		return nil
+	}
+	if order.First {
+		return &sql.ColumnOrder{First: true}
+	} else {
+		return &sql.ColumnOrder{AfterColumn: order.AfterColumn.String()}
+	}
+}
+
+func (b *PlanBuilder) buildIndexDefs(inScope *scope, spec *ast.TableSpec) (idxDefs []*plan.IndexDefinition) {
+	for _, idxDef := range spec.Indexes {
+		constraint := sql.IndexConstraint_None
+		if idxDef.Info.Primary {
+			constraint = sql.IndexConstraint_Primary
+		} else if idxDef.Info.Unique {
+			constraint = sql.IndexConstraint_Unique
+		} else if idxDef.Info.Spatial {
+			constraint = sql.IndexConstraint_Spatial
+		} else if idxDef.Info.Fulltext {
+			// TODO: We do not support FULLTEXT indexes or keys
+			err := sql.ErrUnsupportedFeature.New("fulltext keys are unsupported")
+			b.handleErr(err)
+		}
+
+		columns := b.gatherIndexColumns(idxDef.Columns)
+
+		var comment string
+		for _, option := range idxDef.Options {
+			if strings.ToLower(option.Name) == strings.ToLower(ast.KeywordString(ast.COMMENT_KEYWORD)) {
+				comment = string(option.Value.Val)
+			}
+		}
+		idxDefs = append(idxDefs, &plan.IndexDefinition{
+			IndexName:  idxDef.Info.Name.String(),
+			Using:      sql.IndexUsing_Default, //TODO: add vitess support for USING
+			Constraint: constraint,
+			Columns:    columns,
+			Comment:    comment,
+		})
+	}
+
+	for _, colDef := range spec.Columns {
+		if colDef.Type.KeyOpt == colKeyFulltextKey {
+			err := sql.ErrUnsupportedFeature.New("fulltext keys are unsupported")
+			b.handleErr(err)
+		}
+		if colDef.Type.KeyOpt == colKeyUnique || colDef.Type.KeyOpt == colKeyUniqueKey {
+			idxDefs = append(idxDefs, &plan.IndexDefinition{
+				IndexName:  "",
+				Using:      sql.IndexUsing_Default,
+				Constraint: sql.IndexConstraint_Unique,
+				Comment:    "",
+				Columns: []sql.IndexColumn{{
+					Name:   colDef.Name.String(),
+					Length: 0,
+				}},
+			})
+		}
+	}
+	return
+}
+
+type namedConstraint struct {
+	name string
+}
+
+func (b *PlanBuilder) buildConstraintDefinition(inScope *scope, cd *ast.ConstraintDefinition) interface{} {
```

Comment:
Same comment about being thoughtful with verbs applies throughout here, going to stop making it